tasks/configure.yml

```diff
@@ -152,10 +158,38 @@
     owner: "{{ postgresql_service_user }}"
     group: "{{ postgresql_service_group }}"
     mode: u=rwX,g=rwXs,o=rx
-  notify: restart postgresql with service
+  notify: restart postgresql
+
+- name: PostgreSQL | Enable service
+  service:
+    name: "{{ postgresql_service_name }}"
+    enabled: yes
+  when: postgresql_service_enabled
+
+- name: PostgreSQL | Ensure PostgreSQL is running
+  service:
+    name: "{{ postgresql_service_name }}"
+    state: started
+  register: postgresql_cold_started
 
 - name: PostgreSQL | Reload all conf files
   service:
     name: "{{ postgresql_service_name }}"
     state: reloaded
-  when: postgresql_configuration_pt1.changed or postgresql_configuration_pt2.changed or postgresql_configuration_pt3.changed or postgresql_systemd_custom_conf.changed
+  notify: reload postgresql
```

Comment:
If this is pre v9.5, then a simple reload won't work.  We also need to notify the `reload postgresql` handler so that it can decide if a restart is also necessary.  This reload here is just to prepare the environment to test *IF* it actually needs restarting.